autotest/pst_from_tests.py

```diff
@@ -4576,6 +4577,184 @@ def shortname_conversion_test(tmp_path):
+                        geostruct=grid_gs, 
+                        par_name_base=f.split('.')[1].replace("_","")+"gr", 
+                        pargp=f.split('.')[1].replace("_","")+"gr", 
+                        lower_bound=0.2, upper_bound=5.0, 
+                        ult_ubound=100, ult_lbound=0.01 
+                    )    
+        # pilot point (medium) scale parameters
+        df_pp = pf.add_parameters(f,
+                            zone_array=ib[layer],
+                            par_type="pilotpoints",
+                            use_pp_zones=True,
+                            geostruct=grid_gs,
+                            par_name_base=f.split('.')[1].replace("_","")+"pp",
+                            pargp=f.split('.')[1].replace("_","")+"pp",
+                            lower_bound=0.2,upper_bound=5.0,
+                            ult_ubound=100, ult_lbound=0.01,
+                            pp_space=500) # `
+
+    tag = "sfr_packagedata"
+    files = [f for f in os.listdir(template_ws) if tag in f.lower() and f.endswith(".txt")]
+    f = files[0]
+    # constant and grid scale multiplier conductance parameters
+    name = "sfrcond"
+    df_list = pf.add_parameters(f,
+                    par_type="grid",
+                    geostruct=grid_gs,
+                    par_name_base=name+"gr",
+                    pargp=name+"gr",
+                    index_cols=[0,1,2], # this assumes 1,2 are row,col
+                    use_cols=[8],
+                    lower_bound=0.1,upper_bound=10.0)
+
+    assert df_list.x.max() != df_list.x.min()
+    assert df_list.y.max() != df_list.y.min()
+
+    # add the observations to pf
+    df = pd.read_csv(os.path.join(template_ws, "sfr_obs.csv"), index_col=0)
+    sfr_df = pf.add_observations("sfr_obs.csv",
+                                insfile="sfr_obs.csv.ins", 
+                                index_cols="time", 
+                                use_cols=list(df.columns.values), 
+                                prefix="sfr") 
+
+    pst = pf.build_pst()
+
+    # run once
+    pst.control_data.noptmax=0
+    pst.write(os.path.join(template_ws, 'test.pst'))
+    os_utils.run("{0} test.pst".format(pp_exe_path), cwd=template_ws)
+    pstchk = pyemu.Pst(os.path.join(template_ws,'test.pst'))
+    assert pstchk.phi < 1e-6
+
+
+    # check zone par bounds are respected
+    par = pst.parameter_data
+    assert 'zone' in par.columns
+
+    par['zone'] = [int(i.split(':')[-1]) for i in par.parnme.values]
+    par_org = par.copy()
+    #check gr pars
+    for zone in par.zone.unique():
+        par.loc[(par.zone==zone) & (par.ptype=='gr'), 'parval1'] = float(zone)
+        par.loc[(par.zone==zone) & (par.ptype=='gr'), 'parlbnd'] = float(zone)-0.1
+        par.loc[(par.zone==zone) & (par.ptype=='gr'), 'parubnd'] = float(zone)+0.1
+
+    # write model input files
+    pst.write_input_files(template_ws)
+    pyemu.os_utils.run(r'python forward_run.py', cwd=template_ws)
```

Comment:
It looks like the mf6 run within forward_run.py is failing. Possibly the changes to parval1, above, move values beyond stable values for the forward run? Does it run locally with these values?

Reply:
Hey @briochh, sorry about the hassle...the refactored `pst_from_tests.py` pytests run locally...I've added a few fixes to some other issues that were failing locally...but none of them seem to be related to the failing autotest tests.

Reply:
Looks like you got it @rhugman-intera. Good job dealing with the moving goal posts.